src/connections/DelayedJob.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect } from "react";
+import { createSelector } from "reselect";
+
+import { bulkUpdateJobs, listDelayedJobs } from "@/generated/v3/jobService/jobServiceComponents";
+import {
+  bulkUpdateJobsFetchFailed,
+  bulkUpdateJobsIsFetching,
+  listDelayedJobsFetchFailed
+} from "@/generated/v3/jobService/jobServicePredicates";
+import { DelayedJobData, DelayedJobDto } from "@/generated/v3/jobService/jobServiceSchemas";
+import { ApiDataStore } from "@/store/apiSlice";
+import { Connection } from "@/types/connection";
+import { connectionHook, connectionLoader } from "@/utils/connectionShortcuts";
+
+// --- Delayed Jobs Connection ---
+type DelayedJobsConnection = {
+  delayedJobs?: DelayedJobDto[];
+  isLoading: boolean;
+  hasLoadFailed: boolean;
+};
+
+const delayedJobsSelector = (store: ApiDataStore) => {
+  const delayedJobsMap = store.delayedJobs || {};
+  return Object.values(delayedJobsMap).map(resource => resource.attributes);
+};
+
+const delayedJobsLoadFailedSelector = (store: ApiDataStore) => {
+  return listDelayedJobsFetchFailed(store) != null;
+};
+
+const connectionIsLoaded = ({ delayedJobs, hasLoadFailed, isLoading }: DelayedJobsConnection) => {
+  return (delayedJobs != null && delayedJobs.length > 0) || hasLoadFailed || isLoading;
+};
+
+const delayedJobsConnection: Connection<DelayedJobsConnection> = {
+  load: connection => {
+    const isLoaded = connectionIsLoaded(connection);
+    if (!isLoaded) {
+      listDelayedJobs();
+    }
+  },
+  isLoaded: connectionIsLoaded,
+  selector: createSelector(
+    [delayedJobsSelector, store => delayedJobsLoadFailedSelector(store)],
+    (delayedJobs, hasLoadFailed): DelayedJobsConnection => ({
```

Comment:
🎨 It's OK to leave it, but I believe setting the return type here is not necessary (the type comes from the connection shape... if your return type didn't match DelayedJobsConnection, it would complain even if you don't specify it here):
```
(delayedJobs, hasLoadFailed) => ({
```

Reply:
improved